docs/kb/semgrep-ci/scan-monorepo-in-parts.md

```diff
@@ -0,0 +1,54 @@
+---
+description: How to scan a monorepo in parts for better CI performance and clearer findings organization
+tags:
+  - monorepo
+  - optimization
+  - performance
+  - ci
+---
+
+# Scanning a monorepo in parts
+
+Monorepos, by definition, contain a large amount of code. The code is also often divided into to different components or modules, despite all existing in the same repository.
+
+As such, it can be helpful to scan a monorepo in parts for multiple reasons:
+
+1. Improving scan performance in CI and reducing CI run times
+2. Logically splitting up the monorepo into multiple components to help with managing findings
+
+## How to configure Semgrep in CI to split up a monorepo
+
+Typically when scanning any repo with Semgrep in CI, the base command is `semgrep ci`. Of course, you need to have an auth token set and perhaps other environment variables configured depending on what SCM and CI provider you are using, but this article will assume you are already familiar with those settings.
+
+To split up your monorepo, you need to first decide *how* you want to logically split up the code. For example, if your monorepo has four main modules and their paths are:
+
+    /src/moduleA
+    /src/moduleB
+    /src/moduleC
+    /src/moduleD
+
+Then it would logically make sense to split this into four separate scans in CI, one for each module.
+
+To actually achieve this, you would use the `--include` flag ([see CLI reference](/docs/cli-reference/)) to only scan files in each module's code path:
+
+    semgrep ci --include=/src/moduleA/*
+
+Now, Semgrep is only scanning files under that path and the entire CI run will take less time, since less code is being scanned. Of course, you will now have to run multiple CI runs for the remaining three modules to get full monorepo coverage, but you have more flexibility on how and when you perform those runs.
```

Comment:
I think I'd go ahead and add all four commands under the first one here, something like this:

> For the other modules, the commands look similar:
> 
> &lt;block of commands>
> 
> Each CI run will take less time, and you will have the flexibility to trigger each one on appropriate events or frequencies.
> 

(this isn't ideal wording, here I'm just giving an idea of a different approach)

Reply:
I'm not sure, I think that might make the article unnecessarily convoluted. It's clear they need to run this for the other modules, but when and how they do so is up to them.